agent/src/main/scala/za/co/absa/atum/agent/AtumAgent.scala

```diff
@@ -45,25 +56,36 @@ class AtumAgent private[agent] () {
   /**
    *  Provides an AtumContext given a `AtumPartitions` instance. Retrieves the data from AtumService API.
    *
-   *  @param atumPartitions
-   *  @return
+   * Note: if partitioning doesn't exist in the store yet, a new one will be created with the author stored in
+   *    `AtumAgent.currentUser`. If partitioning already exists, this attribute will be ignored because there
+   *    already is an author who previously created the partitioning in the data store. Each Atum Context thus
+   *    can have different author potentially.
+   *
+   *  @param atumPartitions: Partitioning based on which an Atum Context will be created or obtained.
+   *  @return Atum context object that's either newly created in the data store, or obtained because the input
+   *          partitioning already existed.
    */
   def getOrCreateAtumContext(atumPartitions: AtumPartitions): AtumContext = {
-    val partitioningDTO = PartitioningDTO(AtumPartitions.toSeqPartitionDTO(atumPartitions), None)
+    val authorIfNew = AtumAgent.currentUser
+    val partitioningDTO = PartitioningSubmitDTO(AtumPartitions.toSeqPartitionDTO(atumPartitions), None, authorIfNew)
+
     val atumContextDTO = dispatcher.getOrCreateAtumContext(partitioningDTO)
     lazy val atumContext = AtumContext.fromDTO(atumContextDTO, this)
+
     getExistingOrNewContext(atumPartitions, atumContext)
   }
 
   def getOrCreateAtumSubContext(subPartitions: AtumPartitions)(implicit parentAtumContext: AtumContext): AtumContext = {
+    val authorIfNew = AtumAgent.currentUser
     val newPartitions: AtumPartitions = parentAtumContext.atumPartitions ++ subPartitions
 
     val newPartitionsDTO = AtumPartitions.toSeqPartitionDTO(newPartitions)
     val parentPartitionsDTO = Some(AtumPartitions.toSeqPartitionDTO(parentAtumContext.atumPartitions))
-    val partitioningDTO = PartitioningDTO(newPartitionsDTO, parentPartitionsDTO)
+    val partitioningDTO = PartitioningSubmitDTO(newPartitionsDTO, parentPartitionsDTO, authorIfNew)
 
     val atumContextDTO = dispatcher.getOrCreateAtumContext(partitioningDTO)
     lazy val atumContext = AtumContext.fromDTO(atumContextDTO, this)
```

Comment:
Why is this lazy @salamonpavel ?
If it's do avoid unnecessary creation, it would be enough to send directly `AtumContext.fromDTO(atumContextDTO, this)` into `getExistingOrNewContext`.
`lazy` has too many other undesired consequences.

Reply:
I remember I was asked to refactor the AtumContext.fromDTO into a val before passing into the function call. I agree with the suggested improvement. In this particular case which undesired consequences are realistic in your opinion?